src/net/tcp_stream.rs

```diff
@@ -81,7 +82,12 @@ pub fn spawn_connect_thread(
                 tls,
                 verify_cert,
             } = connection;
-            if !session.connect(&host, port, tls, verify_cert) {
+            let tls_validation = if verify_cert {
```

Comment:
Can we perhaps do a `impl From<bool> for CertificatValidation` here?

Reply:
:+1: Implemented in my recent force push.